src/test_c2s_srv.c

```diff
@@ -35,13 +35,25 @@
  */
 int raw_read(Connection *conn, char* buff, size_t buff_size,
              ssize_t *bytes_read) {
+  int ssl_error;
+  int ssl_errno;
   if (conn->ssl == NULL) {
     *bytes_read = read(conn->socket, buff, buff_size);
     if (*bytes_read <= -1) return errno;
   } else {
     // TODO: Keep track of the number of SSL renegotiations that occur
+    ERR_clear_error();
     *bytes_read = SSL_read(conn->ssl, buff, buff_size);
-    if (*bytes_read <= -1) return SSL_get_error(conn->ssl, *bytes_read);
+    if (*bytes_read <= 0) {
+      ssl_errno = errno;
```

Comment:
I understand that the assignment is unnecessary for non-error cases. My argument for moving this outside the conditional block is for consistency of the pattern "clear error; ssl operation ; ssl_errno = errno" used in other functions. The symmetry makes reasoning about the code a little easier / faster I think.

Would you be willing to move this outside the conditional block after the `SSL_read`?